lib/browser/api/menu.js

```diff
@@ -164,7 +164,8 @@ Menu.buildFromTemplate = function (template) {
   const sorted = sortTemplate(filtered)
 
   sorted.forEach((item) => {
-    if (typeof item !== 'object') {
+    if ((item === null) || (typeof item !== 'object') ||
+        (!item.label && !item.role && !item.type)) {
       throw new TypeError('Invalid template for MenuItem')
```

Comment:
Let's make this checks before creating a `Menu` object, right after checking that `template` is an array.
cc @codebytere if there is any reason why we should postpone the checks till after we have a `sorted` array?